float32[] angles
float32[] distances
float32 max_range
